Sources/EvervaultInputs/Model/PaymentCardData+Validation.swift

```diff
@@ -6,57 +6,65 @@ internal extension PaymentCardData {
         return "\(card.expMonth)/\(card.expYear)"
     }
 
-    init(number: String, cvc: String, expiry: String) {
+    init(number: String, cvc: String, expiry: String, fields: EnabledFields = EnabledFields()) {
         let validator = CreditCardValidator(number)
         self.card.type = validator.predictedType
+        
+        if fields.isCardNumberEnabled {
+            var number = validator.string
+            if let type = self.card.type {
+                number = String(number.prefix(type.validNumberLength.last!))
+            }
+            let formattedNumber = CreditCardFormatter.formatCardNumber(number)
 
-        var number = validator.string
-        if let type = self.card.type {
-            number = String(number.prefix(type.validNumberLength.last!))
+            self.card.number = formattedNumber
+            if validator.isValid {
+                self.card.bin = (self.card.type == .amex) ? String(number.prefix(6)) : String(number.prefix(8))
+                self.card.lastFour = String(number.suffix(4))
+            }
         }
-
-        let formattedNumber = CreditCardFormatter.formatCardNumber(number)
-
-        self.card.number = formattedNumber
-        if validator.isValid {
-            self.card.bin = (self.card.type == .amex) ? String(number.prefix(6)) : String(number.prefix(8))
-            self.card.lastFour = String(number.suffix(4))
+        
+        if fields.isCVCEnabled {
+            self.card.cvc = String(cvc.numbers.prefix(CreditCardValidator.maxCvcLength(for: self.card.type)))
         }
-        self.card.cvc = String(cvc.numbers.prefix(CreditCardValidator.maxCvcLength(for: self.card.type)))
 
-        let expiryParts = expiry.split(separator: "/")
-        self.card.expMonth = expiryParts.count > 0 ? String(expiryParts[0]).numbers : ""
-        self.card.expYear = expiryParts.count > 1 ? String(expiryParts[1]).numbers : ""
+        if fields.isExpiryEnabled {
+            let expiryParts = expiry.split(separator: "/")
+            self.card.expMonth = expiryParts.count > 0 ? String(expiryParts[0]).numbers : ""
+            self.card.expYear = expiryParts.count > 1 ? String(expiryParts[1]).numbers : ""
+        }
 
         let monthNumber = Int(self.card.expMonth)
-        let yearNumber = Int(self.card.expYear)
-
+        
         let actualType = validator.actualType
-        self.isValid = actualType != nil && validator.isValid && CreditCardValidator.isValidCvc(cvc: cvc, type: actualType!) && monthNumber != nil && (1...12).contains(monthNumber!) && yearNumber != nil
-        self.isPotentiallyValid = validator.isPotentiallyValid && (monthNumber == nil || (1...12).contains(monthNumber!))
-        self.isEmpty = number.isEmpty
+        let isCardNumberValid = actualType != nil && validator.isValid && fields.isCardNumberEnabled
```

Comment:
Is card number always required? If card number enabled is false, the validation will always fail